components/constellation/frame.rs

```diff
@@ -29,6 +29,9 @@ pub struct Frame {
     /// The pipeline for the current session history entry
     pub pipeline_id: PipelineId,
 
+    /// The id that maps to the history state entry in the script thread
+    pub state_id: HistoryStateId,
```

Comment:
I suppose we could make it optional and have the BC return `null` for `history.state` if the active `state_id` is `None`